assembly/es.ts

```diff
@@ -85,7 +89,7 @@ export function dayOfWeek(year: i32, month: i32, day: i32): i32 {
 // https://github.com/tc39/proposal-temporal/blob/49629f785eee61e9f6641452e01e995f846da3a1/polyfill/lib/ecmascript.mjs#L2164
 function balanceYearMonth(year: i32, month: i32): YM {
   month -= 1;
-  year  += i32(Math.floor(f32(month) / 12.0));
```

Comment:
I had a feeling that when I wrote this code @MaxGraey would come along shortly afterwards and tidy it up! thanks 👍